app/src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt

```diff
@@ -0,0 +1,204 @@
+        // Duration before a press turns into a long press.
+        // Factor 1.5 is needed, otherwise a long press is not safely detected.
+        // See android.test.TouchUtils longClickView
+        val longPressTimeout = (ViewConfiguration.getLongPressTimeout() * 1.5f).toLong()
+        uiController.loopMainThreadForAtLeast(longPressTimeout)
+        return downEvent
+      }
+
+      override fun sendSwipe(uiController: UiController, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
+        return sendLinearDrag(uiController, downMotion, startCoordinates, endCoordinates, precision)
+      }
+
+      private fun sendLinearDrag(uiController: UiController, downMotion: DragViewAction.DownMotionPerformer, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
+        val steps = interpolate(startCoordinates,endCoordinates)
+        val delayBetweenMovements = DRAG_DURATION / steps.size
+        val downEvent = downMotion.perform(uiController, startCoordinates, precision) ?: return Swiper.Status.FAILURE
+        try {
+          steps.forEachIndexed { index, step ->
+            if (!MotionEvents.sendMovement(uiController, downEvent, step)) {
+              MotionEvents.sendCancel(uiController, downEvent)
+              return Swiper.Status.FAILURE
+            }
+            val desiredTime: Long = downEvent.downTime + delayBetweenMovements * index
+            val timeUntilDesired = desiredTime - SystemClock.uptimeMillis()
+            if (timeUntilDesired > 10) {
+              // If the wait time until the next event isn't long enough, skip the wait
+              // and execute the next event.
+              uiController.loopMainThreadForAtLeast(timeUntilDesired)
+            }
+          }
+          // Wait before sending up because some drag handling logic may discard move events
+          // that has been sent immediately before the up event. e.g. HandleView.
+          uiController.loopMainThreadForAtLeast(WAIT_BEFORE_SENDING_UP.toLong())
+          if (!MotionEvents.sendUp(uiController, downEvent, endCoordinates)) {
+            MotionEvents.sendCancel(uiController, downEvent)
+            return Swiper.Status.FAILURE
+          }
+        } finally {
+          downEvent.recycle()
+        }
+        return Swiper.Status.SUCCESS
+      }
+
+      private fun interpolate(start:FloatArray,end:FloatArray):Array<FloatArray>{
+        val res = Array(DRAG_STEP_COUNT) { FloatArray(2) }
+        for (i in 0 until DRAG_STEP_COUNT) {
+          res[i][0] = start[0] + (end[0] - start[0]) * i / (DRAG_STEP_COUNT - 1f)
+          res[i][1] = start[1] + (end[1] - start[1]) * i / (DRAG_STEP_COUNT - 1f)
+        }
+        return res
+      }
+
+      override fun toString(): String { return "long press and drag" }
+    };
+
+    override fun wrapUiController(uiController: UiController): UiController {
```

Comment:
Why is this needed?

Reply:
it was used for the enum pattern. Removed now